### DISCUSSION TITLE
cartesian_coeff values for linear motion?

### DISCUSSION THREAD

**Discussion by mohamedsayed18** [tesseract-robotics/tesseract_ros#132] [Q&A] [answered]
In the `basic_cartesian_example.cpp` 
https://github.com/tesseract-robotics/tesseract_ros/blob/c62960cbeaeb2c29429a40c097a6ffc62762c958/tesseract_ros_examples/src/basic_cartesian_example.cpp#L238-L241
I tried to make the cartesian_coeff equal to zero `plan_profile->cartesian_coeff = Eigen::VectorXd::Constant(6, 1, 0);`, As I understand this means that there is no constrain on the pose(position and orientation) of the generated path. but the linear plan instruction "RASTER" was not generated only the Freespace ones with no error messages. I made two assumptions depending on this which I would like to confirm:
* The `LINEAR` plan instruction was not generated as there are conditions that are going against each other, Linear motion assumes free translation and no rotations but I defined free rotation too.
* If a specific plan instruction is not achievable it will be skipped with no error messages and other plan instructions in the same program will be executed.
Update: I noticed this behavior in Freespace instructions too.

**Comment by Levi-Armstrong** [comment]
It should be in the trajopt default plan profile in tesseract motion
planners.

**Reply by mohamedsayed18** [reply]
Sorry it was my mistake, I deleted my answer, but I'm still trying to understand the behaviour mentioned above

**Comment by Levi-Armstrong** [comment, marked as answer]
I did not follow the comment related to the LINEAR plan instruction. You added a LINEAR plan instruction but set cartesian coeff to zero? This should not be allowed (I assume it is not currently checking this) because then no constraint/cost will be added. It should at least have one non-zero coefficient.

> If a specific plan instruction is not achievable it will be skipped with no error messages and other plan instructions in the same program will be executed.

If they are assigned as costs then this is expected behavior. If they are assigned as constraints and the constraint is not satisfied it should error.
